### PR TITLE
[GEOS-9647] Black lives matter

### PR DESCRIPTION
**Pull request by davidblasby**
update text to be more inclusive

This is for https://osgeo-org.atlassian.net/browse/GEOS-9647

This updates;

* whitelist -> allow-list
* blacklist -> deny-list
* master/slave -> primary/replica
* master (branch) -> main development branch (not in all cases) - github will be changing "master" to "main" in the future.
* master password -> root password

There are a few other changes - like a few typos I found and whitespace->blank-space.

I had issues with the non-english .properties files containing text that didn't seem like UTF-8, so i didn't modify those files.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

For all pull requests:

- [ x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ na] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [na] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: doc/en/user/source/security/passwd.rst [hunk 1]
```diff
@@ -85,23 +85,23 @@ The GeoServer keystore is password protected with a :ref:`security_master_passwd
...
 
 * Protect access to the :ref:`keystore <security_passwd_keystore>`
 * Protect access to the GeoServer :ref:`security_root`
 
-By default, the master password is generated and stored in a file named ``security/masterpw.info`` using plain text. When upgrading from an existing GeoServer data directory (versions 2.1.x and lower), the algorithm attempts to figure out the password of a user with the role ``ROLE_ADMINISTRATOR``. If such a password is found and the password length is 8 characters at minimum, GeoServer uses this password as master password. Again, the name of the chosen user is found in ``security/masterpw.info``. 
+By default, the root password is generated and stored in a file named ``security/masterpw.info`` using plain text. When upgrading from an existing GeoServer data directory (versions 2.1.x and lower), the algorithm attempts to figure out the password of a user with the role ``ROLE_ADMINISTRATOR``. If such a password is found and the password length is 8 characters at minimum, GeoServer uses this password as master password. Again, the name of the chosen user is found in ``security/masterpw.info``.
```
Comment (bradh) [inline doc/en/user/source/security/passwd.rst:96]: There is a second "master password" in this line.

**Review by bradh — COMMENTED** [review]
I'm -0 on the whitespace change. Technically its not just blank spaces, and historically its from the colour of the paper. I don't have the historical or US political perspective though, so I won't block on that.

Other than that, I think the changes are valid, and even without the inclusiveness aspect, they make the document easier to understand (e.g. master/slave only makes sense if you already understood that as a scale-out concept, while primary/replica is more descriptive and obvious).

File: doc/en/user/source/community/jms-cluster/activemq/JDBC.rst [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
+If the primary looses connection to the database or looses the exclusive lock then it immediately shuts down. If a primary shuts down or fails, one of the other replicas will grab the lock and so the topology switches to the following diagram
 
 .. figure:: images/MasterFailed.png
    :align: center
 
-One of the other other slaves immediately grabs the exclusive lock on the database to them commences becoming the master, starting all of its transport connectors.
-Clients loose connection to the stopped master and then the failover transport tries to connect to the available brokers - of which the only one available is the new master.
-Master restart
-At any time you can restart other brokers which join the cluster and start as slaves waiting to become a master if the master is shutdown or a failure occurs. So the following topology is created after a restart of an old master...
+One of the other other replicas immediately grabs the exclusive lock on the database to them commences becoming the primary, starting all of its transport connectors.
```
[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/community/jms-cluster/activemq/JDBC.rst:26]: @iant1212 already hooked up a spell check at some point, I think we are free to fix any mistakes like this as we see them.

File: doc/en/user/source/data/app-schema/tutorial.rst [hunk 3]
```diff
@@ -80,7 +80,7 @@ For ``gsml:MappedFeature`` the file is ``workspaces/gsml/gsml_MappedFeature/data
         </connectionParameters>
     </dataStore>
 
-.. note:: Ensure that there is no whitespace inside an ``entry`` element.
+.. note:: Ensure that there is no blank-space inside an ``entry`` element.
```
Comment (jodygarnett) [inline doc/en/user/source/data/app-schema/tutorial.rst:83]: I think we are stuck with whitespace characters for a bit, part of the unicode standard etc. I spent some time researching and could not find any evidence of white space (a visual design concept) being considered in a racist context.

File: doc/en/developer/source/programming-guide/security/index.rst [hunk 4]
```diff
@@ -182,11 +182,11 @@ Password Policy
 ``PasswordPolicy`` is the interface for validating user passwords, applying constraints such
 as password length, character mix, etc...
 
-MasterPasswordProvider
+RootPasswordProvider
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Security service that provides a method for obtaining the GeoServer master password.
-The master password serves two purposes.
+Security service that provides a method for obtaining the GeoServer root password.
```
[the remark this answers is not in the document]
Comment (davidblasby) [reply doc/en/developer/source/programming-guide/security/index.rst:188]: I can change to "keystore password"

File: doc/en/user/source/community/jms-cluster/activemq/SharedFolder.rst [hunk 5]
```diff
@@ -44,22 +44,22 @@ Clients should be using the Failover Transport to connect to the available broke
...
-If the master looses the exclusive lock then it immediately shuts down. If a master shuts down or fails, one of the other slaves will grab the lock and so the topology switches to the following diagram
+If the primary looses the exclusive lock then it immediately shuts down. If a primary shuts down or fails, one of the other replicas will grab the lock and so the topology switches to the following diagram
 
 .. figure:: images/MasterFailed.png
    :align: center
 
-One of the other other slaves immediately grabs the exclusive lock on the file system to them commences becoming the master, starting all of its transport connectors.
-Clients loose connection to the stopped master and then the failover transport tries to connect to the available brokers - of which the only one available is the new master.
+One of the other other replicas immediately grabs the exclusive lock on the file system to them commences becoming the primary, starting all of its transport connectors.
+Clients loose connection to the stopped primary and then the failover transport tries to connect to the available brokers - of which the only one available is the new primary.
```
[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/community/jms-cluster/activemq/SharedFolder.rst:57]: @bradh do you have the ability as a geoserver team member to direct edit PR? I find it helpful for quickly correcting documentation sphinx mistakes before merging.

File: doc/en/developer/source/release-guide/index.rst [hunk 6]
```diff
@@ -135,23 +135,23 @@ When creating the first release candidate of a series, there are some extra step
...
      * ``doc/en/user/source/conf.py``
 
-* Commit the changes and push to the master branch on GitHub::
+* Commit the changes and push to the the main development branch branch on GitHub::
 
       git commit -am "Updated version to 2.12-SNAPSHOT, updated GeoTools dependency to 18-SNAPSHOT, updated GeoWebCache dependency to 1.12-SNAPSHOT, and related changes"
       git push geoserver master
       
-* Create the new RC version in `JIRA <https://osgeo-org.atlassian.net/projects/GEOS>`_ for issues on master; for example, if master is now ``2.12-SNAPSHOT``, create a Jira version ``2.12-RC1`` for the first release of the ``2.12.x`` series
+* Create the new RC version in `JIRA <https://osgeo-org.atlassian.net/projects/GEOS>`_ for issues on the main development branch; for example, if the primary development branch is now ``2.12-SNAPSHOT``, create a Jira version ``2.12-RC1`` for the first release of the ``2.12.x`` series
```
[the remark this answers is not in the document]
Comment (davidblasby) [reply doc/en/developer/source/release-guide/index.rst:143]: yes - I noticed I did that a few times...  I think github is going to move from "master" to "main", so "main development branch" is better.

File: doc/en/user/source/data/app-schema/tutorial.rst (hunk 3 above)
Comment (davidblasby) [reply doc/en/user/source/data/app-schema/tutorial.rst:83]: I just made the change due to its proximity to the "safe space" term.

**Comment by jodygarnett** [comment]
Thanks @davidblasby and @bradh for working through this discussion.

File: doc/en/user/source/community/jms-cluster/activemq/JDBC.rst [hunk 7]
```diff
@@ -1,43 +1,43 @@
...
------------------------------
+Configuring JDBC Primary/Replica
+--------------------------------
 
 
 .. figure:: images/MasterRestarted.png
    :align: center
 
-By default if you use the <jdbcPersistenceAdapter/> to avoid the high performance journal you will be using JDBC Master Slave by default. You just need to run more than one broker and point the client side URIs to them to get master/slave. This works because they both try an acquire an exclusive lock on a shared table in the database and only one will succeed.
+By default if you use the <jdbcPersistenceAdapter/> to avoid the high performance journal you will be using JDBC Primary/Replica by default. You just need to run more than one broker and point the client side URIs to them to get primary/replica. This works because they both try an acquire an exclusive lock on a shared table in the database and only one will succeed.
```
[the remark this answers is not in the document]
Comment (davidblasby) [reply doc/en/user/source/community/jms-cluster/activemq/JDBC.rst:38]: I think this is correct

[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/community/jms-cluster/activemq/JDBC.rst:38]: The sentence is correct `them` refers to the more than one broker

**Comment by jodygarnett** [comment]
A test failure, likely based on UI test checking for "master" password:
```
org.geoserver.security.web.passwd.MasterPasswordChangePanelTest.testBadCurrentPassword(MasterPasswordChangePanelTest.java:54)
```

**Comment by jodygarnett** [comment]
Thanks @davidblasby, going to squash this into a single commit so we can get the "[GEOS-9647]  ..." format.

**Pull request merged**